hv.c

```diff
@@ -648,11 +648,11 @@ Perl_hv_common(pTHX_ HV *hv, SV *keysv, const char *key, STRLEN klen,
 
                 if (keysv || is_utf8) {
                     if (!keysv) {
-                        keysv = newSVpvn_utf8(key, klen, TRUE);
+                        keysv = newSVpvn_flags(key, klen, SVf_UTF8|SVs_TEMP);
                     } else {
-                        keysv = newSVsv(keysv);
+                        keysv = sv_2mortal(newSVsv(keysv));
```

Comment:
Shouldn't this use `sv_mortalcopy_flags(keysv, SV_GMAGIC|SV_NOSTEAL)` instead?

(I really wish sv_mortalcopy() and newSVsv() used the same flags /cry)

Reply:
> Shouldn't this use `sv_mortalcopy_flags(keysv, SV_GMAGIC|SV_NOSTEAL)` instead?

It should do in the following commit? (I could squash the two together.)

> (I really wish sv_mortalcopy() and newSVsv() used the same flags /cry)

`sv_mortalcopy` & `newSVsv_flags` being almost the same (apart from the mortalisation) but not quite doing the same checks makes me sad. I keep thinking about merging them, but the differences always puts me off.

Reply:
> It should do in the following commit? (I could squash the two together.)

No it's fine as it is, sometimes I focus too much on one commit and miss the final results.